irohad/torii/impl/command_service_transport_grpc.cpp

```diff
@@ -72,55 +78,81 @@ namespace torii {
+                        &error) {
+                  status_bus_->publish(status_factory_->makeStatelessFail(
+                      error.error.hash, error.error.error));
+                  return false;
+                });
+          })
+        | boost::adaptors::transformed([&](auto result) {
+            return std::move(
+                       boost::get<iroha::expected::ValueOf<decltype(result)>>(
+                           result))
+                .value;
+          }));
+  }
+
   grpc::Status CommandServiceTransportGrpc::ListTorii(
       grpc::ServerContext *context,
       const iroha::protocol::TxList *request,
       google::protobuf::Empty *response) {
-    auto tx_list_builder = shared_model::proto::TransportBuilder<
-        shared_model::interface::TransactionSequence,
-        shared_model::validation::DefaultUnsignedTransactionsValidator>();
-
-    tx_list_builder.build(*request).match(
-        [this](
-            iroha::expected::Value<shared_model::interface::TransactionSequence>
-                &tx_sequence) {
-          this->command_service_->handleTransactionList(tx_sequence.value);
-        },
-        [this, request](auto &error) {
-          auto &txs = request->transactions();
-          if (txs.empty()) {
-            log_->warn("Received no transactions. Skipping");
-            return;
-          }
-          using HashType = shared_model::crypto::Hash;
-
-          std::vector<HashType> hashes;
-          std::transform(
-              txs.begin(),
-              txs.end(),
-              std::back_inserter(hashes),
-              [](const auto &tx) {
-                return shared_model::crypto::DefaultHashProvider::makeHash(
-                    shared_model::proto::makeBlob(tx.payload()));
+    auto transactions = deserializeTransactions(request);
+
+    auto batches = batch_parser_->parseBatches(transactions);
+
+    for (auto &batch : batches) {
+      shared_model::interface::TransactionBatchFactory::createTransactionBatch(
+          batch,
+          shared_model::validation::DefaultUnsignedTransactionsValidator())
+          .match(
+              [&](iroha::expected::Value<
+                  shared_model::interface::TransactionBatch> &value) {
+                this->command_service_->handleTransactionBatch(
+                    std::make_shared<shared_model::interface::TransactionBatch>(
+                        std::move(value).value));
+              },
+              [&](iroha::expected::Error<std::string> &error) {
+                std::vector<shared_model::crypto::Hash> hashes;
+
+                std::transform(batch.begin(),
+                               batch.end(),
+                               std::back_inserter(hashes),
+                               [](const auto &tx) { return tx->hash(); });
+
+                auto error_msg = formErrorMessage(hashes, error.error);
+                // set error response for each transaction in a sequence
```

Comment:
Maybe change "sequence" to "batch candidate"? So not to confuse those who know, what the tx_sequence was